packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -427,6 +427,21 @@ export class DataStores implements IDisposable {
             throw responseToException(create404Response(request), request);
         }
 
+        if (context.tombstoned) {
+            // The requested data store is tombstoned by gc. Throw a 404 response exception.
+            const request = { url: id };
+            const urlNoQuery = request.url.split("?")[0];
+            const errWithStack = generateErrorWithStack();
+
+            const response: IResponse = {
```

Comment:
Can you use `createResponseError` helper function? To avoid repeating things like the mimeType, stack getter, and consistent formatting of the error message.